doc/api/net.md

```diff
@@ -515,56 +544,74 @@ changes:
     description: The `hints` option is supported now.
 -->
 
-Opens the connection for a given socket.
-
-For TCP sockets, `options` argument should be an object which specifies:
-
-  - `port`: Port the client should connect to (Required).
-
-  - `host`: Host the client should connect to. Defaults to `'localhost'`.
+* `options` {Object}
+* `connectListener` {Function} Common parameters of [`socket.connect()`][]
+  methods. Will be added as a listener for the [`'connect'`][] event once.
+* Returns: {net.Socket} The socket itself.
 
-  - `localAddress`: Local interface to bind to for network connections.
+Initiate the connection on a given socket.
 
-  - `localPort`: Local port to bind to for network connections.
+Note: this method is similar to a [connect(2)][] call, and before that,
```

Comment:
maybe describe the calls in the order they would be made, instead of the reverse order? I'm also not sure that this is useful, we don't usually describe the complex series of system calls required to set things up (cf. the `ipc` option in child process `spawn()`). Are you hoping to direct people to where they can get more information if the node docs are insufficient to understand how the API works? If so, people haven't liked that approach in the fs docs.